### ISSUE TITLE
Recurrence for club items?

### ISSUE THREAD

**Issue by ericberman** [ericberman/MyFlightbookWeb#775]
Probably would want to load a library for this, but it means having a way to edit one vs. edit many, support for various recurrence rules, and exceptions.

**Comment by fkrauthan** [comment]
I was actually wondering the same if there is an easy way to tag airplanes and then use airplane tags in custom currency rules. This would make managing my club currencies a lot easier.

**Comment by ericberman** [comment]
Tag how?  Custom currency rules already allow you to limit to specific aircraft; what would tagging do that this wouldn't?

**Comment by fkrauthan** [comment]
One of the clubs I fly with has more then 25 planes. So it would be a lot easier to just add to each plane when I add it `PFC` for example as club and don't have to adjust the custom currency rule all the time to include more planes (as that list becomes very quickly very hard to manage). It would allow me to just say I have to fly any aircraft with tag `PFC` within the last 60 days as my custom currency rule.

**Comment by ericberman** [comment]
I guess I don't see how adding each plane to a currency rule is more work then editing the private notes for each plane.  It's actually less: you can do it all at once, on one screen.  With tagging each aircraft, you have to go into each aircraft, edit the private notes, hit "update", go to the next aircraft, do the same, ...  It's a lot more steps.

Private notes, by the way, is a great place to put tags (I generally recommend using a "#" prefix/suffix simply to avoid false positives/negatives) and are used when searching, but my currency computation code doesn't pull that in to the data it examines.

**Comment by fkrauthan** [comment]
Well my reasoning is the following: I don't add all planes right away anyways so I add them as I fly them. Right now I have todo 3 steps:

1. Create the plane
2. Update my currency rule
3. Save my flight

If I could tag (or a simple category) my planes I would save one step.

1. Create the plane (including its tags/category)
2. Save my flight

Especially since the list is pretty big of planes its not very user friendly to manage to be honest (on the currency screen) but that are just my two cents.

**Comment by ericberman** [comment]
For the tags - I think the private note will do what you want.  If you set up the currency rule with "Contains Text" and put that text into the private notes for the aircraft, it will automatically be picked up. Again, I recommend a unique prefix/suffix like "#" to avoid false positives/negatives, so you could use (for example) "#G1000#" for Garmin G1000 equipped aircraft.  I think that does exactly what you want.  

So allow me to close that part of the issue.  This remains recurrence.

**Comment by fkrauthan** [comment]
So to make sure I understand it. If I put for example `#PFC#` into the Private note of a plane and then put `#PFC#` into the `Contains this text` field on my custom currency it will pickup that plane properly?

**Comment by ericberman** [comment]
Yes, it should pick up that *flight* correctly.  (Currency looks at flights, but in doing so, it also looks at the private notes for the aircraft used in a flight).

**Comment by fkrauthan** [comment]
@ericberman it does not seem to work. I just created a flight for today with `C-FCAP` which has as private note `#PFC#` but when using that in my custom currency rule `Pacific Flying Club - 60 Day Currency` it does not show up anymore.

**Comment by ericberman** [comment]
We are WAAAAAY off of recurrence for club items.  Let's please move this to issue #880.

**Comment by ericberman** [comment]
Thoughts for implementation: can greatly simplify if...

1. Require explicit number of recurrences or duration
2. Create distinct item for each recurrence.

E.g., you say "3rd thursday of each month, through 12/1/2022".  Create however many match that.

Simplifies things because:
 * No need for exceptions
 * No need for "Delete all"
